packages/itwin/grouping-mapping-widget/src/widget/components/CustomCalculationTable.tsx

```diff
@@ -3,57 +3,51 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 import {
-  SvgAdd,
   SvgDelete,
   SvgEdit,
   SvgMore,
 } from "@itwin/itwinui-icons-react";
 import {
-  Button,
   DropdownMenu,
   IconButton,
   MenuItem,
   Table,
 } from "@itwin/itwinui-react";
 import React, { useMemo, useState } from "react";
 import type { CreateTypeFromInterface } from "../utils";
-import { PropertyMenuView } from "./PropertyMenu";
 import type { CellProps } from "react-table";
 import DeleteModal from "./DeleteModal";
 import type { CustomCalculation } from "@itwin/insights-client";
 import { useMappingClient } from "./context/MappingClientContext";
 import { useGroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import { PropertyTableToolbar } from "./PropertyTableToolbar";
+import { PropertyNameCell } from "./PropertyNameCell";
 
-export type ICustomCalculationTyped =
+type ICustomCalculationTyped =
   CreateTypeFromInterface<CustomCalculation>;
 
 interface CustomCalculationTableProps {
-  iModelId: string;
   mappingId: string;
   groupId: string;
-  setSelectedCustomCalculation: React.Dispatch<React.SetStateAction<ICustomCalculationTyped | undefined>>;
-  setGroupModifyView: React.Dispatch<React.SetStateAction<PropertyMenuView>>;
-  onCustomCalculationModify: (value: CellProps<ICustomCalculationTyped>) => void;
+  onClickAddCustomCalculationProperty?: () => void;
+  onClickModifyCustomCalculation?: (value: CustomCalculation) => void;
   isLoadingCustomCalculations: boolean;
-  customCalculations: ICustomCalculationTyped[];
+  customCalculations: CustomCalculation[];
   refreshCustomCalculations: () => Promise<void>;
-  selectedCustomCalculation?: ICustomCalculationTyped;
 }
 
-const CustomCalculationTable = ({
-  iModelId,
+export const CustomCalculationTable = ({
   mappingId,
   groupId,
-  setSelectedCustomCalculation,
-  setGroupModifyView,
-  onCustomCalculationModify,
+  onClickAddCustomCalculationProperty,
+  onClickModifyCustomCalculation,
   isLoadingCustomCalculations,
   customCalculations,
   refreshCustomCalculations,
-  selectedCustomCalculation,
 }: CustomCalculationTableProps) => {
-  const { getAccessToken } = useGroupingMappingApiConfig();
+  const { getAccessToken, iModelId } = useGroupingMappingApiConfig();
   const mappingClient = useMappingClient();
+  const [selectedCustomCalculation, setSelectedCustomCalculation] = useState<CustomCalculation | undefined>(undefined);
   const [
     showCustomCalculationDeleteModal,
     setShowCustomCalculationDeleteModal,
```

Comment:
The state name is too verbose. `showDeleteModal` would be enough. There is only one delete modal in this component.